Extend Docker docs with invoke on each runtime

jldec opened the pull request:
Proposal to document both runtimes and make the getting started flow a bit more elucidating.
If this looks ok, i'll apply similar to the other getting started pages.
thx.

jldec committed:
Extend Docker docs with invoke on each runtime

jldec committed:
fix column names

scothis commented on docs/v0.4/getting-started/docker-for-mac.md:
```diff
@@ -6,16 +6,17 @@ sidebar_label: Docker for Mac
 The following will help you get started running a riff function with Knative on Docker Community Edition for Mac.
 
-### TL;DR
+## TL;DR
```
Can we drop the tl;dr now that the pagenav has similar content (with links)?

jldec replied:
👍 yes - was thinking the same.

scothis commented on docs/v0.4/getting-started/docker-for-mac.md:
````diff
@@ -160,34 +160,101 @@ NAME     LATEST IMAGE
+The ClusterIP service created by a deployer is not reachable from outside the cluster, so it is easier to create a new NodePort service.
 
+```sh
+kubectl expose deployment k8s-square-deployer \
````
`expose` creates a new Kubernetes service. `kubectl port-forward` is a better way to create a transient connection to the function. For a permanent access, ingress should be setup for the cluster.

jldec replied:
ok - with change to port-forward

scothis commented on docs/v0.4/getting-started/docker-for-mac.md:
````diff
@@ -160,34 +160,101 @@ NAME     LATEST IMAGE
 ```
 
-## create a deployer
+## create a Knative deployer
````
Can we link to the core/knative runtime docs instead?

jldec replied:
This is a tricky balance between using a one-page "getting started" to convey the concept of multiple runtimes which is one of the biggest new ideas for this release of riff, or erring on the side of shorter, just to get users to a place where something works.

I'm actually tempted to go in the other direction and show BOTH an app workload with the Core runtime, and a function workload with the Knative runtime.

cc @markfisher

jldec replied:
In the interest of shipping the docs, resolved to leave as-is for now, and do similar flow for the other 3 environments. 

One idea to consider for the future would be to have a single opinionated getting-started page wih just 1 environment and 1 runtime to create and deploy 1 function, and then also provide dedicated pages for installation in each environment, as well as pages for each workload type (functions, applications, containers).

scothis commented on docs/v0.4/getting-started/docker-for-mac.md:
````diff
@@ -160,34 +160,101 @@ NAME     LATEST IMAGE
 
-Deployers take a built function and deploy it to a riff runtime. There are two runtimes: core and knative. The core runtime is always available, while the knatve runtime is only available on clusters with Istio and Knative installed.
+```sh
+curl http://localhost:$INGRESS_PORT/ -w '\n' \
````
How is this port exposed on localhost?

jldec replied:
it's on a nodePort since we installed Istio with` --set gateways.istio-ingressgateway.type=NodePort`

jldec committed:
Address review comments
- drop TL;DR
- use `kubectl port-forward` with Core Runtime deployer

also remove extra screenshot for validating kubectl context

jldec committed:
Refine port-forward to use auto-assigned port

jldec committed:
Change port-forward to match Core runtime page
